### ISSUE TITLE
Nginx configuration file syntax is not ok : sudo: sorry, you must have a tty to run sudo

### ISSUE THREAD

**Issue by shuangxi588** [jslsolucoes/nginx-admin#17]
I already added  user nginx-agent in sudoers file.

1.
# tail -n 1 /etc/sudoers
nginx-agent ALL=(ALL) NOPASSWD:/usr/sbin/nginx,/usr/bin/pgrep nginx,/usr/bin/killall nginx

2.
# su - nginx-agent 
Last login: Mon Feb 26 14:30:45 CST 2018 on pts/3
su: warning: cannot change directory to /home/nginx-agent: No such file or directory
-bash-4.2$ sudo nginx -t
nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
-bash-4.2$ exit

**Comment by shuangxi588** [comment]
![image](https://user-images.githubusercontent.com/30979619/36656172-fee97c6c-1b01-11e8-8c1f-70638b6b9d4b.png)

**Comment by shuangxi588** [comment]
![image](https://user-images.githubusercontent.com/30979619/36656205-26b740a8-1b02-11e8-8b91-a23b684104eb.png)

**Comment by riclava** [comment]
use visudo and comment "Default requiretty" may help you ;-)

**Comment by shuangxi588** [comment]
Thanks.
![image](https://user-images.githubusercontent.com/30979619/36657261-d6ca006c-1b06-11e8-957a-07e6990cb8a0.png)

**Comment by shuangxi588** [comment]
Thanks for your help, but I still have two questions.
1. can i change database use maridb instead of mysql.
2. i want to use the default nginx directory "/etc/nginx"  instead of  nginx-agent directory "/opt/nginx-agent-2.0.1/settings"   ??

**Comment by riclava** [comment]
@shuangxi588 
1. I guess it will works well as you use mariadb
2. it can be, but you should not do like that because you need to change tpl and some other code like:
```
user  nginx;
worker_processes  1;

error_log  /opt/nginx-admin/settings/log/error.log warn;
pid        /opt/nginx-admin/settings/nginx.pid;

```
and
```
	<#if virtualHost.https == 1>
		listen               443 ssl;
		ssl_certificate      /opt/nginx-admin/settings/ssl/${ virtualHost.sslCertificate.resourceIdentifierCertificate.hash };
		ssl_certificate_key  /opt/nginx-admin/settings/ssl/${ virtualHost.sslCertificate.resourceIdentifierCertificatePrivateKey.hash };
		......
        ssl_prefer_server_ciphers on;
        ssl_session_cache shared:SSL:10m;
        ssl_dhparam  /opt/nginx-admin/settings/ssl/dhparam.pem;
	<#else>
	 	listen               80;
	</#if>
```

**Comment by shuangxi588** [comment]
OK,thanks very much.

**Comment by jslsolucoes** [comment]
@shuangxi588 

Disable/Comment entire tty its not best way to do that. 

You can add single user directive to disable tty when activated using:  
Defaults:nginx-agent !requiretty 

To use mariadb as database you can follow mysql steps i guess (not tested yet):

#login as root and create database and user
create database nginx_admin;
grant all privileges on nginx_admin.* to nginx_admin@'%' identified by 'password';
flush privileges;
		
#Go to nginx-admin folder default installation /opt/nginx-admin/conf/nginx-admin.conf 
#and change settings to your database settings like :
NGINX_ADMIN_DB_DRIVER=mysql
NGINX_ADMIN_DB_HOST=localhost
NGINX_ADMIN_DB_PORT=3306
NGINX_ADMIN_DB_NAME=nginx_admin
NGINX_ADMIN_DB_USERNAME=nginx_admin
NGINX_ADMIN_DB_PASSWORD=password

#Restart service. After that your database will be installed on mariadb with default username and password
service nginx-admin restart 

Can you try and give some feedback ?

**Comment by jslsolucoes** [comment]
Added official support to mariadb.
Added directive to avoid tty in nginx-admin-agent user script install. 
Will be released to the 2.0.2 with new documentation site
